Panic when trying to create a PR to a repo without write access

### Describe the bug

gh version 0.6.3 (2020-03-31)
https://github.com/cli/cli/releases/tag/v0.6.3

I tried sending a PR to a repo before I had write access to it:

```
my-branch> gh pr create
panic: value method github.com/cli/cli/api.Repository.RepoOwner called using nil *Repository pointer

goroutine 1 [running]:
github.com/cli/cli/api.(*Repository).RepoOwner(0x0, 0xc000854480, 0xc)
	<autogenerated>:1 +0x97
github.com/cli/cli/internal/ghrepo.IsSame(0x1addc40, 0xc000830160, 0x1addc40, 0x0, 0x0)
	/home/runner/work/cli/cli/internal/ghrepo/repo.go:54 +0x5d
github.com/cli/cli/command.prCreate(0x2017b60, 0x204f078, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/cli/cli/command/pr_create.go:157 +0x27a1
github.com/spf13/cobra.(*Command).execute(0x2017b60, 0x204f078, 0x0, 0x0, 0x2017b60, 0x204f078)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:840 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0x2015400, 0x5, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v0.0.6/command.go:945 +0x317
main.main()
	/home/runner/work/cli/cli/cmd/gh/main.go:32 +0xde
```

After being given push access, the error went away:

```
my-branch> gh pr create
Creating pull request for my-branch into master in dummy/dummy-repo

? Title (this is a test)
...
```

### Steps to reproduce the behavior

1. Clone any repo that you don't have push access to 
2. Create a new branch
3. type `gh pr create`

### Expected vs actual behavior

Expected:
A user error, suggesting a lack of write access

Actual:
Panic (as shown above)

### Logs

See above.

--- a/api/fake_http.go
+++ b/api/fake_http.go
@@ -48,6 +48,10 @@ func (f *FakeHTTP) StubWithFixture(status int, fixtureFileName string) func() {
 }
 
 func (f *FakeHTTP) StubRepoResponse(owner, repo string) {
+	f.StubRepoResponseWithPermission(owner, repo, "WRITE")
+}
+
+func (f *FakeHTTP) StubRepoResponseWithPermission(owner, repo, permission string) {
 	body := bytes.NewBufferString(fmt.Sprintf(`
 		{ "data": { "repo_000": {
 			"id": "REPOID",
@@ -56,9 +60,9 @@ func (f *FakeHTTP) StubRepoResponse(owner, repo string) {
 			"defaultBranchRef": {
 				"name": "master"
 			},
-			"viewerPermission": "WRITE"
+			"viewerPermission": "%s"
 		} } }
-	`, repo, owner))
+	`, repo, owner, permission))
 	resp := &http.Response{
 		StatusCode: 200,
 		Body:       ioutil.NopCloser(body),

--- a/command/pr_create.go
+++ b/command/pr_create.go
@@ -95,7 +95,9 @@ func prCreate(cmd *cobra.Command, _ []string) error {
 
 	// otherwise, determine the head repository with info obtained from the API
 	if headRepo == nil {
-		headRepo, _ = repoContext.HeadRepo()
+		if r, err := repoContext.HeadRepo(); err == nil {
+			headRepo = r
+		}
 	}
 
 	baseBranch, err := cmd.Flags().GetString("base")
@@ -236,6 +238,10 @@ func prCreate(cmd *cobra.Command, _ []string) error {
 		headBranchLabel = fmt.Sprintf("%s:%s", headRepo.RepoOwner(), headBranch)
 	}
 
+	if headRemote == nil {
+		headRemote, _ = repoContext.RemoteForRepo(headRepo)
+	}
+
 	// There are two cases when an existing remote for the head repo will be
 	// missing:
 	// 1. the head repo was just created by auto-forking;
@@ -243,7 +249,7 @@ func prCreate(cmd *cobra.Command, _ []string) error {
 	//
 	// In either case, we want to add the head repo as a new git remote so we
 	// can push to it.
-	if err != nil {
+	if headRemote == nil {
 		// TODO: support non-HTTPS git remote URLs
 		headRepoURL := fmt.Sprintf("https://github.com/%s.git", ghrepo.FullName(headRepo))
 		// TODO: prevent clashes with another remote of a same name
@@ -260,13 +266,6 @@ func prCreate(cmd *cobra.Command, _ []string) error {
 
 	// automatically push the branch if it hasn't been pushed anywhere yet
 	if headBranchPushedTo == nil {
-		if headRemote == nil {
-			headRemote, err = repoContext.RemoteForRepo(headRepo)
-			if err != nil {
-				return fmt.Errorf("git remote not found for head repository: %w", err)
-			}
-		}
-
 		pushTries := 0
 		maxPushTries := 3
 		for {

--- a/command/pr_create_test.go
+++ b/command/pr_create_test.go
@@ -65,6 +65,49 @@ func TestPRCreate(t *testing.T) {
 	eq(t, output.String(), "https://github.com/OWNER/REPO/pull/12\n")
 }
 
+func TestPRCreate_withForking(t *testing.T) {
+	initBlankContext("OWNER/REPO", "feature")
+	http := initFakeHTTP()
+	http.StubRepoResponseWithPermission("OWNER", "REPO", "READ")
+	http.StubResponse(200, bytes.NewBufferString(`
+		{ "data": { "repository": { "forks": { "nodes": [
+		] } } } }
+	`))
+	http.StubResponse(200, bytes.NewBufferString(`
+		{ "data": { "repository": { "pullRequests": { "nodes" : [
+		] } } } }
+	`))
+	http.StubResponse(200, bytes.NewBufferString(`
+		{ "node_id": "NODEID",
+		"name": "REPO",
+		"owner": {"login": "myself"},
+		"clone_url": "http://example.com",
+		"created_at": "2008-02-25T20:21:40Z"
+		}
+	`))
+	http.StubResponse(200, bytes.NewBufferString(`
+		{ "data": { "createPullRequest": { "pullRequest": {
+			"URL": "https://github.com/OWNER/REPO/pull/12"
+		} } } }
+	`))
+
+	cs, cmdTeardown := test.InitCmdStubber()
+	defer cmdTeardown()
+
+	cs.Stub("")                                         // git config --get-regexp (determineTrackingBranch)
+	cs.Stub("")                                         // git show-ref --verify   (determineTrackingBranch)
+	cs.Stub("")                                         // git status
+	cs.Stub("1234567890,commit 0\n2345678901,commit 1") // git log
+	cs.Stub("")                                         // git remote add
+	cs.Stub("")                                         // git push
+
+	output, err := RunCommand(prCreateCmd, `pr create -t title -b body`)
+	eq(t, err, nil)
+
+	eq(t, http.Requests[3].URL.Path, "/repos/OWNER/REPO/forks")
+	eq(t, output.String(), "https://github.com/OWNER/REPO/pull/12\n")
+}
+
 func TestPRCreate_alreadyExists(t *testing.T) {
 	initBlankContext("OWNER/REPO", "feature")
 	http := initFakeHTTP()

